File: zantt-api-backend/src/main/java/com/zantt/entities/ProjectEntity.java
package com.zantt.entities;

import javax.persistence.*;

@Entity
@Table(name = "projects")
public class ProjectEntity {
    @Id
    @Column(name = "project_id")
    private String projectId;
    @Column(name = "name", nullable = false)
    private String name;

    public String getProjectId() {
        return projectId;
    }

    public void setProjectId(String projectId) {
        this.projectId = projectId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: zantt-api-backend-net/Zantt/Entities/ProjectEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Zantt.Entities;

[Table("projects")]
public class ProjectEntity
{
    [Key]
    [Column("project_id")]
    [StringLength(255)]
    public string ProjectId { get; set; } = Guid.NewGuid().ToString();

    [Required]
    [Column("name")]
    [StringLength(255)]
    public string Name { get; set; } = string.Empty;
}
